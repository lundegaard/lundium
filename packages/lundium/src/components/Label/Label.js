import React from 'react';
import PropTypes from 'prop-types';
import { cx } from 'ramda-extension';

import { useTheme } from '../ThemeProvider';

const Label = ({
	children,
	className,
	isHidden,
	disabled,
	readOnly,
	...rest
}) => {
	const theme = useTheme();

	return (
		<label
			{...rest}
			className={cx(
				'control-label',
				{
					'control-label--dir-rtl': theme.isRTL,
					['sr-only']: isHidden,
					'control-label--readonly': readOnly,
					'control-label--disabled': disabled,
				},
				className,
			)}
		>
			{children}
		</label>
	);
};

Label.propTypes = {
	/** Children to be rendered in the main container. */
	children: PropTypes.node,
	/** Class name for root. */
	className: PropTypes.string,
	disabled: PropTypes.bool,
	/** Whether the label should be hidden on regular screens. */
	isHidden: PropTypes.bool,
	readOnly: PropTypes.bool,
};

export default Label;
