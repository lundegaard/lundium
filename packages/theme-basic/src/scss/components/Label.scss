@import '../core';

/*
 -  Form label
 -  (<label class="control-label" />)
 */


.control-label {
	position: absolute;
	inset: logical 0 0 auto auto;
	// padding-left: unset;
	// padding-right: unset;
	padding-inline-start: calc( #{$input-padding-x} + #{$input-border-width} );
	padding-inline-end: 0;
	font-size: $control-label-font-size;
	font-weight: 600;
	// letter-spacing: .5px;

	&--disabled,
	&--readonly {
		color: $control-label-disabled-color;
	}
}
