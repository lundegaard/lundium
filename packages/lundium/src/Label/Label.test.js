import React from 'react';
import { shallow } from 'enzyme';

import Label from './Label';

describe('<Label />', () => {
	const requiredProps = {};

	it('matches snapshot when shallowly rendered', () => {
		const wrapper = shallow(<Label {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
