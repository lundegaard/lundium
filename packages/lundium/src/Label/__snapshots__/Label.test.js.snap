// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> matches snapshot when shallowly rendered 1`] = `
<label
  className="control-label"
/>
`;
